eclog/src/test/java/com/ericsson/bss/cassandra/ecaudit/eclog/config/TestEcLogYamlConfigLoader.java

```diff
@@ -0,0 +1,132 @@
+
+/**
+ * Tests the {@link EcLogYamlConfigLoader} class.
+ */
+public class TestEcLogYamlConfigLoader
+{
+    private static final Path LOG_DIR = Paths.get("/log");
+    private static final Path CUSTOM_FILE = Paths.get("/custom/config.yaml");
+    private static final Path WORKING_DIR = Paths.get(System.getProperty("user.dir"));
+
+    private static final String EXPECTED_FILE_NAME = "ecLog.yaml";
+    private static final Path CONFIG_FILE_DIR = Paths.get("src/test/resources/");
+    private static final Path CONFIG_FILE = CONFIG_FILE_DIR.resolve(EXPECTED_FILE_NAME);
+
+    @Test
+    public void testLoadWithoutConfigFileGivesDefaultValues()
+    {
+        ToolOptions options = ToolOptions.builder().withPath(LOG_DIR).build();
+        EcLogYamlConfig config = EcLogYamlConfigLoader.load(options);
+
+        assertDefaultConfig(config);
+    }
+
+    @Test
+    public void testLoadWithConfigFromFileInLogDirPath()
+    {
+        ToolOptions options = ToolOptions.builder().withPath(CONFIG_FILE_DIR).build();
+        EcLogYamlConfig config = EcLogYamlConfigLoader.load(options);
+
+        assertCustomConfig(config);
+    }
+
+    @Test
+    public void testLoadWithConfigFileProvidedOnCommandLine()
+    {
+        ToolOptions options = ToolOptions.builder().withPath(LOG_DIR).withConfig(CONFIG_FILE).build();
+        EcLogYamlConfig config = EcLogYamlConfigLoader.load(options);
+
+        assertCustomConfig(config);
+    }
+
+    @Test
+    public void testConfigFileLocations()
+    {
+        ToolOptions options = ToolOptions.builder().withPath(LOG_DIR).build();
+        List<Path> locations = EcLogYamlConfigLoader.getConfigFileLocations(options);
+
+        assertThat(locations).containsExactly(WORKING_DIR.resolve(EXPECTED_FILE_NAME),
+                                              LOG_DIR.resolve(EXPECTED_FILE_NAME));
+    }
+
+    @Test
+    public void testConfigFileLocationsIncludingCommandLineConfig()
+    {
+        ToolOptions options = ToolOptions.builder().withPath(LOG_DIR).withConfig(CUSTOM_FILE).build();
+        List<Path> locations = EcLogYamlConfigLoader.getConfigFileLocations(options);
+
+        assertThat(locations).containsExactly(CUSTOM_FILE,
+                                              WORKING_DIR.resolve(EXPECTED_FILE_NAME),
+                                              LOG_DIR.resolve(EXPECTED_FILE_NAME));
+    }
+
+    @Test
+    public void testLoadEmptyConfigFromUrlGivesDefaultConfig()
+    {
+        Path path = Paths.get("src/test/resources/empty.yaml");
+        EcLogYamlConfig config = EcLogYamlConfigLoader.loadConfigFromUrl(path);
+
+        assertDefaultConfig(config);
+    }
+
+    @Test
+    public void testLoadConfigFromInvalidUrl()
+    {
+        Path path = Paths.get("invalid_path");
+        assertThatExceptionOfType(IllegalArgumentException.class)
+        .isThrownBy(() -> EcLogYamlConfigLoader.loadConfigFromUrl(path))
+        .withMessage("Invalid configuration file: invalid_path");
```

Comment:
This isn't likelly to happen in practice (I think) since we are doing filter(Files::isReadable) in load().